IdentityCore/src/oauth2/ciam/MSIDCIAMIdTokenClaims.m

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+#if !EXCLUDE_FROM_MSALCPP
+
+#import "MSIDCIAMIdTokenClaims.h"
+#import "MSIDHelpers.h"
+#import "MSIDCIAMAuthority.h"
+
+@implementation MSIDCIAMIdTokenClaims
+
+- (void)initDerivedProperties
+{
+    [super initDerivedProperties];
+
+    // Set userId
+    NSString *userId = self.preferredUsername;
```

Comment:
Isn't this duplicate code with what is in AAD? If yes, is it needed?

Reply:
it's a bit different from what is in MSIDAADv2TokenIdClaims. The CIAMIdTokenClaims was adapted from MSIDB2CIdTokenClaims

Reply:
It seems to work without a separate MSIDCIAMIdTokenClaims class, so I removed it. Adjusted code to use MSIDAADV2IdTokenClaims class instead.